### PR TITLE
xhyve: update with version that fixes thread lock issues on some OSes

### PR DESCRIPTION
**Pull request by aphor**
A newer rev of xhyve is available which fixes the Sierra/pre-Sierra thread lock issues.
Changing the version to this rev brings in fixes for Sierra.

https://github.com/mist64/xhyve/commit/1f1dbe3059904f885e4ab2b3328f4bb350ea5c37

Also install the manpage.

### CODE REVIEW

**Comment by tobypeterson** [comment]
Is this a specifically recommended commit, or just the latest? Do you have any info on the next planned actual release?

**Comment by aphor** [comment]
It's the latest. Xhyve is usable, but still fairly experimental. I've
tested it, and read through the latest commits solving the Sierra spin lock
deprecation warnings/errors.

If I were in charge of the xhyve tags, I'd tag this release. I'm happy with
it on Sierra. So, I recommend this release over the version in the current
Portfile. When the upstream tags a new version, I'll test that and submit
another pull request.

File: emulators/xhyve/Portfile [hunk 1]
```diff
@@ -3,8 +3,8 @@
 PortSystem          1.0
 PortGroup           github 1.0
 
-github.setup        mist64 xhyve 0.2.0 v
-revision            2
+github.setup        mist64 xhyve 1f1dbe3 
+revision            0
```
Comment (mojca) [inline emulators/xhyve/Portfile]: You may delete the whole line setting the revision to zero as that's the default anyway.

But you have to add an explicit version, else there will be troubles with incrementing the version number. You may use something like
```
version 0.2.1-git-1f1dbe3
```
(just making things up).

**Comment by mojca** [comment]
Did this start working on pre-Yosemite as well?

You should prefix the commit messages with the `portname:` and you could merge the two commits together in this case and force push to the branch.

File: emulators/xhyve/Portfile (hunk 1 above)
Comment (aphor) [inline emulators/xhyve/Portfile]: I don't control the upstream xhyve repo, so I can't provide version tags.

I forgot that Macports needs versions to be increasing; subsequent hashes will appear sometimes to Macports as downgrades.

I'll open an issue with @mist64 to tag a new version, but in the meantime, will the automatic git adapter parse the hash out of the version string like 1f1dbe3 from 0.2.1-git-1f1dbe3 ??

**Comment by aphor** [comment]
@mojca xhyve depends on the MacOS Hypervisor.framework which is new in Yosemite.

- [MacOS Yosemite Release Notes](https://developer.apple.com/library/content/releasenotes/MacOSX/WhatsNewInOSX/Articles/MacOSX10_10.html)
- [MacOS Hypervisor API Reference](https://developer.apple.com/reference/hypervisor) 

Maybe it could be backported to older versions of Darwin? If so rather than OS version checking, maybe a framework dependency check would be better?

**Comment by mojca** [comment]
Unless (until) somebody has shown a successful (and legal?) porting of the framework, it makes no sense to assume that it will work. I only asked because your commit removed
```tcl
if {${os.platform} eq "darwin" && (${os.major} < 14 ...) } {
    # error
}
```
suggesting that the new version should work on < 10.10.

> I don't control the upstream xhyve repo, so I can't provide version tags.

I didn't ask you to. It's certainly helpful to encourage upstream to tag useful releases, but you don't need them to do anything if all you want is to use some untagged commit for a valid reason.

> ... but in the meantime, will the automatic git adapter parse the hash out of the version string like 1f1dbe3 from 0.2.1-git-1f1dbe3 ??

No, that was probably a misunderstanding. You need to use both of the following lines (you just need to fix the version as seen by MacPorts):
```tcl
github.setup        mist64 xhyve 1f1dbe3
version             0.2.1-git-1f1dbe3
```
The second line is just a suggestion. If you plan to upgrade to a different tag from 0.2.1 later, the above suggestion is still bad and you should better add a date of commit, for example `0.2.1-20160809` or `0.2.1-20160809-1f1dbe3` or `0.2.1-git-20160809-1f1dbe3` or basically anything you want as long as you can ensure increasing version numbers. I would also say "the shorter the better" (but unambiguous) for aesthetical reasons (to avoid super long lines in `port installed` etc.).

**Comment by mojca** [comment]
Just a bit of a technical rant.
The last commit "merge 'master' into master" is a bit weird. (It's generally not a good idea to create a pull request from your master branch. You should ideally keep the master clean and only add commits related to one particular topic in a specific branch. We try to keep the history linear and do rebasing.)

**Comment by aphor** [comment]
Sorry, I think the last changes to master obviate my PR and I did the merge to my branch out of habit to review the differences. I got sidetracked before I could kill this PR though, so sorry, and thanks.

**Pull request closed without merging**